pkg/cri/server/service.go

```diff
@@ -347,6 +342,17 @@ func (c *criService) register(s *grpc.Server) error {
 	return nil
 }
 
+// getSandboxController returns the sandbox controller configuration for sandbox.
+// If absent in legacy case, it will return the default controller.
+func (c *criService) getSandboxController(config *runtime.PodSandboxConfig, runtimeHandler string) (sandbox.Controller, error) {
+	ociRuntime, err := c.getSandboxRuntime(config, runtimeHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sandbox runtime: %w", err)
+	}
+
+	return c.client.SandboxController(ociRuntime.Sandboxer), nil
```

Comment:
This is causing a unit test regression as some of the unit tests do not have the client defined. We should figure out a way to avoid calling the client here.

Reply:
I'm sorry I don't quite get it, when I run `make test`, all tests passed, and we can see many places in cri codes to call client, why should we avoid calling client here? I think I can make it by defining a `sandboxControllers` in `criService`, but I don't get the reason why we have to do it. Could you please tell me why?

Reply:
I see, it is the TestListContainerStats that need to get platform from sandbox controller

Reply:
Maybe I can define a `sandboxService` interface in cri and define the `Platform()` method in it, and I can define a `fakeSandboxService` struct in test codes and init such an object into the `criService` in `newTestCRIService`.

Reply:
Also opened #9268 to fix the test. The direction we are going through is like you have it, reducing use of the client.

Reply:
I made a fix in #9275, please take a look.